volume/store/store.go

```diff
@@ -309,16 +309,20 @@ func (s *VolumeStore) create(name, driverName string, opts, labels map[string]st
 // This makes sure there are no races between checking for the existence of a volume and adding a reference for it
 func (s *VolumeStore) GetWithRef(name, driverName, ref string) (volume.Volume, error) {
 	name = normaliseVolumeName(name)
```

Comment:
And I think you're right, maybe there are many "slow" locks in other places. It is a good idea to fix them in another PR.